osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd

```diff
@@ -1238,13 +1238,60 @@
     <xsd:complexContent>
       <xsd:extension base="common:Command">
         <xsd:sequence>
-          <xsd:element name="directAttach" type="xsd:int"/>
-          <xsd:element name="randomisationStartWindow" type="xsd:int"/>
-          <xsd:element name="multiplicationFactor" type="xsd:int"/>
-          <xsd:element name="numberOfRetries" type="xsd:int"/>
+          <xsd:element name="directAttach" type="xsd:int" />
+          <xsd:element name="randomisationStartWindow" type="xsd:int" />
+          <xsd:element name="multiplicationFactor" type="xsd:int" />
+          <xsd:element name="numberOfRetries" type="xsd:int" />
         </xsd:sequence>
       </xsd:extension>
     </xsd:complexContent>
   </xsd:complexType>
 
+  <xsd:element name="GetKeysRequest">
+    <xsd:complexType>
+      <xsd:sequence>
+        <xsd:element type="tns:SecretType" name="secretType"
+          minOccurs="1" maxOccurs="4" />
+      </xsd:sequence>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="GetKeysAsyncRequest">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="common:AsyncRequest" />
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="GetKeysResponse">
+    <xsd:complexType>
+      <xsd:sequence>
+        <xsd:element type="tns:SecretType" name="secretType" />
+        <xsd:element type="xsd:hexBinary" name="secretValue" />
+      </xsd:sequence>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="GetUnencryptedKeysAsyncResponse">
```

Comment:
The renames removing "Unencrypted" from the elements related to get keys requests and responses look good.
The only thing is they are incomplete at this location. Changing `GetUnencryptedKeysAsyncResponse` to `GetKeysAsyncResponse` would also fix what is now a validation error in `SmartMeteringConfiguration.wsdl` at line 448.